### PR TITLE
Objectify and improve structuring

### PR DESCRIPTION
**Pull request by kedhammar**

### CODE REVIEW

File: anglerfish/demux/demux.py [hunk 1]
```diff
@@ -163,119 +184,180 @@ def layout_matches(
...
+        if (
+            len(
+                [
+                    i
+                    for i, j in enumerate(entries_index_dists)
+                    if j == entries_index_dists[entries_min_index_dist_loc]
+                ]
+            )
+            > 1
+        ):
```
Comment (remiolsen) [inline anglerfish/demux/demux.py:320]: I'm not sure this is more readable than the previous version if I'm honest. At least it's too spaced out for taste.

**Review by remiolsen — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -187,3 +187,12 @@ Also, the [Anglerfish logo](docs/Anglerfish_logo.svg) was designed by [@FranBona
 <p align="center">
   <img src="docs/Anglerfish_logo.svg">
 </p>
+
+### GitHub contributors
```
[the remark this answers is not in the document]
Comment (kedhammar) [reply README.md]: 3c1c9da61233830be1d557f14ce1d90cad80e808

File: anglerfish/demux/demux.py (hunk 1 above)
Comment (kedhammar) [reply anglerfish/demux/demux.py]: I tried doing some invasive simplification of the code, hope it tastes better

[23f3166](https://github.com/remiolsen/anglerfish/pull/86/commits/23f316625c438a3f1010a97d37c9d9d0602305ef)

**Review by remiolsen — APPROVED** [review]

**Review by alneberg — APPROVED** [review]
Looks good! Can't say I followed on all the changes but didn't find anything weird and it seems to run still. We need to decide on the use of adapter vs adaptor though. It seems to be roughly 50/50 at the moment but we use them synonymously right?

**Pull request merged**